### ISSUE TITLE
Segmentation fault on startup after building on Ubuntu Studio 19.10

### ISSUE THREAD

**Issue by tomb7890** [hydrogen-music/hydrogen#804]
Hydrogen version *          : e317548515bfc64b973f873b47fa6c1f8df73297
Operating system + version  : Ubuntu Studio 19.10
Audio driver + version      :

Audio:     Device-1: Intel Xeon E3-1200 v3/4th Gen Core Processor HD Audio driver: snd_hda_intel
           Device-2: Intel 9 Series Family HD Audio driver: snd_hda_intel
           Device-3: VIA ICE1712 [Envy24] PCI Multi-Channel I/O driver: snd_ice1712
           Device-4: Logitech HD Pro Webcam C920 type: USB driver: snd-usb-audio,uvcvideo
           Sound Server: ALSA v: k5.3.0-40-lowlatency

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Hi, I am new to this code base and newish to Hydrogen app. Trying to build and run the latest.  Some gdb output appears below. Before this, I'd been running Ubuntu Studio and its included binary of hydrogen 0.9.7 for about a week without issues. 
```
tomb@fractaldm:~/code/hydrogen$ ./build/src/gui/hydrogen
qt5ct: using qt5ct plugin

Hydrogen 1.0.0-'e3175485' [Feb 23 2020]  [http://www.hydrogen-music.org]
Copyright 2002-2008 Alessandro Cominu
Copyright 2008-2019 The hydrogen development team
Hydrogen comes with ABSOLUTELY NO WARRANTY
This is free software, and you are welcome to redistribute it under certain conditions. See the file COPYING for details.
QPainter::begin: Paint device returned engine == 0, type: 2
QPainter::setFont: Painter not active
QPainter::setPen: Painter not active
QPainter::end: Painter not active, aborted
qt5ct: palette support is disabled
qt5ct: custom style sheet is disabled
(E) Filesystem::check_permissions /usr/local/share/hydrogen/data/ is not a directory
(E) Filesystem::bootstrap will use local data path : /home/tomb/code/hydrogen/build/src/gui/data/
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/ is not a directory
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/click.wav is not a file
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/DefaultSong.h2song is not a file
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/demo_songs/ is not a directory
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/drumkits/ is not a directory
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/emptySample.wav is not a file
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/hydrogen.default.conf is not a file
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/i18n/ is not a directory
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/img/ is not a directory
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/xsd/ is not a directory
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/xsd/drumkit_pattern.xsd is not a file
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/xsd/drumkit.xsd is not a file
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/xsd/playlist.xsd is not a file
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/hydrogen.default.conf is not a file
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/click.wav is not a file
(E) Sample::load Unable to read /home/tomb/code/hydrogen/build/src/gui/data/click.wav
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/emptySample.wav is not a file
(E) Sample::load Unable to read /home/tomb/code/hydrogen/build/src/gui/data/emptySample.wav
(E) Filesystem::check_permissions /home/tomb/code/hydrogen/build/src/gui/data/emptySample.wav is not a file
(E) Sample::load Unable to read /home/tomb/code/hydrogen/build/src/gui/data/emptySample.wav
(E) SongReader::getPath Song file /home/tomb/code/hydrogen/build/src/gui/data/DefaultSong.h2song not found.
Segmentation fault (core dumped)
tomb@fractaldm:~/code/hydrogen$
```

```
Thread 1 "hydrogen" received signal SIGSEGV, Segmentation fault.
0x000055555565e260 in InstrumentEditor::InstrumentEditor (this=0x555555e2d720, pParent=0x0) at /home/tomb/code/hydrogen/src/gui/src/InstrumentEditor/InstrumentEditor.cpp:499
499                   m_nSelectedComponent = pComponentList->front()->get_id();

```

**Comment by tomb7890** [comment]
Jack Messages:

```14:22:49.771 ALSA connection change.
Sun Feb 23 14:22:49 2020: New client 'Hydrogen' with PID 11484
Sun Feb 23 14:22:49 2020: ERROR: Unknown destination port in attempted (dis)connection src_name [Hydrogen:out_L] dst_name [alsa_pcm:playback_1]
Sun Feb 23 14:22:49 2020: Connecting 'Hydrogen:out_L' to 'system:playback_1'
Sun Feb 23 14:22:49 2020: Connecting 'Hydrogen:out_R' to 'system:playback_2'
Sun Feb 23 14:22:49 2020: ERROR: JackEngine::XRun: client = Hydrogen was not finished, state = Triggered
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: Disconnecting 'Hydrogen:out_L' from 'system:playback_1'
Sun Feb 23 14:22:49 2020: Disconnecting 'Hydrogen:out_R' from 'system:playback_2'
Sun Feb 23 14:22:49 2020: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sun Feb 23 14:22:49 2020: ERROR: Failed to find port 'Hydrogen:out_L' to destroy
Sun Feb 23 14:22:49 2020: ERROR: Failed to find port 'Hydrogen:out_R' to destroy
Sun Feb 23 14:22:49 2020: Client 'Hydrogen' with PID 11484 is out

```

**Comment by cme** [comment]
Hi, this is just because the paths are set up to find resources starting from the directory of the hydrogen executable. It can't be run directly from build/src/gui as it won't find its resources through there:

> (E) Filesystem::bootstrap will use local data path : /home/tomb/code/hydrogen/build/src/gui/data/

The "build.sh" script in the top level directory of the repo copies the executable to the top level, and it should be run from there:

> katana~/hydrogen ./build.sh  mm
> [ ... build output ...]
> [100%] Built target hydrogen
> katana~/hydrogen ./hydrogen 
> 
> Hydrogen 1.0.0-'d29d9e0b' [May 25 2020]  [http://www.hydrogen-music.org]
> Copyright 2002-2008 Alessandro Cominu
> Copyright 2008-2020 The hydrogen development team
> Hydrogen comes with ABSOLUTELY NO WARRANTY
> This is free software, and you are welcome to redistribute it under certain conditions. See the file  COPYING for details.
> (E) Filesystem::check_permissions /usr/local/share/hydrogen/data/ is not a directory
> (E) Filesystem::bootstrap will use local data path : /home/colin/hydrogen/data/

Can you try that and see if it works for you?

**Comment by cme** [comment]
The actual *crash* in this case should have been fixed by #1155. Closing.
